src/ProductConstructionService/ProductConstructionService.Cli/Operations/DeploymentOperation.cs

```diff
@@ -126,30 +126,38 @@ public async Task<int> RunAsync()
         }
     }
 
-    private async Task CleanupRevisionsAsync(IEnumerable<ContainerAppRevisionTrafficWeight> revisionsTrafficWeight)
+    private async Task RemoveRevisionLabel(string revisionName, string? label)
     {
-        // Cleanup all revision labels
-        foreach (var revisionTrafficWeight in revisionsTrafficWeight)
+        if (!string.IsNullOrEmpty(label))
         {
-            if (!string.IsNullOrEmpty(revisionTrafficWeight.Label))
-            {
-                var result = await InvokeAzCLI([
-                        "containerapp", "revision", "label", "remove",
+            var result = await InvokeAzCLI([
+                    "containerapp", "revision", "label", "remove",
                     ],
-                    [
-                        "--label", revisionTrafficWeight.Label
-                    ]);
-                result.ThrowIfFailed($"Failed to remove label {revisionTrafficWeight.Label} from revision {revisionTrafficWeight.RevisionName}. Stderr: {result.StandardError}");
-            }
+                [
+                    "--label", label
+                ]);
+            result.ThrowIfFailed($"Failed to remove label {label} from revision {revisionName}. Stderr: {result.StandardError}");
         }
+    }
 
-        // Now deactivate all revisions in the list
-        foreach (var revisionTrafficWeight in revisionsTrafficWeight)
+    private async Task CleanupRevisionsAsync(IEnumerable<ContainerAppRevisionTrafficWeight> revisionsTrafficWeight)
+    {
+        List<ContainerAppRevisionResource> activeRevisions = _containerApp.GetContainerAppRevisions()
+            .ToEnumerable()
+            .Where(revision => revision.Data.IsActive == true)
+            .Where(revision => revision.Data.TrafficWeight != 100)
```

Comment:
```suggestion
            .Where(revision => revision.Data.IsActive)
            .Where(revision => revision.Data.TrafficWeight != 100)
```

Reply:
Or

```suggestion
            .Where(revision => revision.Data.IsActive && revision.Data.TrafficWeight != 100)
```

Reply:
`IsActive` is a `bool?` so had to do a something else. I kind of like having 2 `Wheres` in two lines, feels nicer to look at, I think the compiler does optimization there